R/derive_date_vars.R

```diff
@@ -69,6 +69,18 @@
 #' cut off date. Only dates which are in the range of possible dates are
 #' considered.
 #'
+#' @param preserve Preserve partial dates when doing date imputation for middle
```

Comment:
The `preserve` parameter should affect all imputations not only `date_imputation = "mid"`.

Reply:
As I was finishing this up - I was like maybe we will want to use this preserve function more?  Should this be created as a separate issue as this one just focus on "mid" in this issue?

> * In the "Breaking Changes" section should be mentioned that the behaviour for `date_imputation = "mid"` has been changed.
> * There are some functions which provide the `date_imputation` parameter (`derive_disposition_dt()` and `lstalvdt_source()`). I think they should provide the `preserve` parameter as well.
> * I am not sure what we should do for functions which provide no `date_imputation` parameter but impute dates, e.g., `filter_date_source()`, which is called by `derive_param_tte()`. What do you think?

Hey Stefan.  Many thanks for reviewing.  

- I will update the Breaking Changes section.
- Should this use of the preserve parameter be a separate issue as I'm changing multiple functions now?  Nervous to change so much in one Pull Request.  
- I'm not really sure - perhaps something we should bring up to the wider group?

Reply:
As the `preserve` parameter is already implemented in this issue, I see no benefit in moving it into a separate issue.